src/FEValues/FunctionValues.jl

```diff
@@ -0,0 +1,78 @@
+# Helpers to get the correct types for FunctionValues for the given function and, if needed, geometric interpolations. 
+struct SInterpolationDims{rdim,sdim} end
+struct VInterpolationDims{rdim,sdim,vdim} end
+function InterpolationDims(::ScalarInterpolation, ip_geo::VectorizedInterpolation{sdim}) where sdim
+    return SInterpolationDims{getdim(ip_geo),sdim}()
+end
+function InterpolationDims(::VectorInterpolation{vdim}, ip_geo::VectorizedInterpolation{sdim}) where {vdim,sdim}
+    return VInterpolationDims{getdim(ip_geo),sdim,vdim}()
+end
+
+typeof_N(::Type{T}, ::SInterpolationDims) where T = T
+typeof_N(::Type{T}, ::VInterpolationDims{<:Any,dim,dim}) where {T,dim} = Vec{dim,T}
```

Comment:
Should this be
```suggestion
typeof_N(::Type{T}, ::VInterpolationDims{dim,dim,dim}) where {T,dim} = Vec{dim,T}
```
@termi-official ?

Reply:
The type of the N should be independent of the reference dimension, right?

Reply:
It is not required I think; I thought for consistency since when either `rdim!=sdim or rdim!=vdim`, both dNdx and dNd\xi are StaticArrays (as discussed above). (Just to make sure I didn't change anything regarding the previous implementation).